addons/mail/static/src/models/attachment/attachment_tests.js

```diff
@@ -137,6 +137,28 @@ QUnit.test('isViewable', async function (assert) {
     assert.ok(attachment.isViewable);
 });
 
+QUnit.test('uploadController', async function (assert) {
```

Comment:
Better have test named like an assertion, e.g. _"uploading attachment should have abort controller to abort upload-in-progress"_.

It gives more direction to what we want, how stuffs must be named, and how short the test should be.

Reply:
I just used the same naming as other tests in the file ..

Reply:
Yes, tests in other files were made when we had less experience with defining good test. We thought just having name of field or method in test name was good enough, but in the end it encourages poor tests, and doesn't tell what or why these fields/methods are important.

As time passes, I become more concerned with tests basically reflecting the functional specs. It's tempting to just think of tests as as a measure to validate correctness of code, which is just one goal of a test. The problem to just think of that is that you are very likely to make bad tests. Bad tests make code maintenance more painful, because they could either break or pass easily when they shouldn't, and they are not well documented so it's a big effort to fix them.

IMO, a bad test is much much worse than no test at all. Good code should be easy to read and understand, and tests are there to clearly tell what we want to achieve, and prove it is achieved.